### PR TITLE
build: Add scala-version to matrix

### PR DESCRIPTION
**Pull request by snmvaughan**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/388

## Rationale for this change

scalafix check will fail when running Scala 2.13, but is skipped for Scala 2.12.

## What changes are included in this PR?

Add testing of Scala 2.13 along with 2.12 by adding `scala-version` to the matrix.  Explicitly exclude combinations of Spark `3.2` and Scala `2.13`.

## How are these changes tested?

Local repo

### CODE REVIEW

File: .github/workflows/pr_build.yml [hunk 1]
```diff
@@ -82,11 +83,14 @@ jobs:
...
+        scala-version: ['2.12', '2.13']
         exclude:
           - java_version: 17
             spark-version: '3.2'
           - java_version: 11
             spark-version: '3.2'
+          - spark-version: '3.2'
+            scala-version: '2.13'
       fail-fast: false
     name: ${{ matrix.os }}/java ${{ matrix.java_version }}-spark-${{matrix.spark-version}}/${{ matrix.test-target }}
```
Comment (viirya) [inline .github/workflows/pr_build.yml:95]: Add scala-version to the pipeline name?

Comment (snmvaughan) [reply .github/workflows/pr_build.yml:95]: I missed that.

**Review by viirya — COMMENTED** [review]
Do you want to fix these scala format errors together? Otherwise these new pipelines are not passed.

**Comment by snmvaughan** [comment]
> Do you want to fix these scala format errors together? Otherwise these new pipelines are not passed.

I expected these errors, which is why I wanted to get Scala 2.13 into the pipeline in the first place.  The real question is do we want to add a bunch of unrelated fixes into this PR?

File: spark/src/test/scala/org/apache/comet/CometCastSuite.scala [hunk 2]
```diff
@@ -867,26 +867,27 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
-        (SQLConf.SESSION_LOCAL_TIMEZONE.key, "America/Los_Angeles")) {
-        val df = data.withColumn("converted", col("a").cast(toType))
-        df.collect()
-        val str =
-          new ExtendedExplainInfo().generateExtendedInfo(df.queryExecution.executedPlan)
-        assert(str.contains(expectedMessage))
-      }
-    }
-  }
+  // TODO Commented out to work around scalafix since this is currently unused.
```
Comment (snmvaughan) [inline spark/src/test/scala/org/apache/comet/CometCastSuite.scala:870]: @andygrove I commented this out thinking you still might want it.

**Comment by viirya** [comment]
> I expected these errors, which is why I wanted to get Scala 2.13 into the pipeline in the first place. The real question is do we want to add a bunch of unrelated fixes into this PR?

If we don't fix them here, we will have failed pipelines. I think we should not merge a PR which has failed pipelines.

File: spark/src/test/scala/org/apache/comet/CometCastSuite.scala (hunk 2 above)
Comment (andygrove) [reply spark/src/test/scala/org/apache/comet/CometCastSuite.scala:870]: Thank you. I will see if this is still needed and follow up to use or remove.

File: spark/src/test/scala/org/apache/comet/exec/CometColumnarShuffleSuite.scala [hunk 3]
```diff
@@ -587,7 +587,7 @@ abstract class CometColumnarShuffleSuite extends CometTestBase with AdaptiveSpar
   }
 
   test("columnar shuffle on nested array") {
-    Seq("false", "true").foreach { execEnabled =>
+    Seq("false", "true").foreach { _ =>
```
Comment (andygrove) [inline spark/src/test/scala/org/apache/comet/exec/CometColumnarShuffleSuite.scala:590]: looks like this needs to be addressed in a follow on PR

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @snmvaughan

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Merged. Thanks @snmvaughan @andygrove

File: spark/src/test/scala/org/apache/comet/exec/CometColumnarShuffleSuite.scala (hunk 3 above)
Comment (snmvaughan) [reply spark/src/test/scala/org/apache/comet/exec/CometColumnarShuffleSuite.scala:590]: Good point.  That `execEnabled` wasn't being used.

**Pull request merged**